store/src/java/com/zimbra/cs/html/owasp/OwaspHtmlSanitizer.java

```diff
@@ -57,6 +57,7 @@ private String checkUnbalancedTags(String html) {
         ArrayList<String> tags = new ArrayList<String>();
         // in tags array we can add multiple tags, so for each tag write start &
         // end with a & separator like(<!--&-->)
+        html = html.replaceAll("<!-->","-->");
```

Comment:
we are doing replaceAll on entire html content of mail
- what would be performance implication of this change?
- if user has used <!--> inside mail as content, then what would happen with that data?

Reply:
@silentsakky I do agree on the performance impact, but if we see below there're many string operations going on
Ideally, as I understood this issue should be fixed from the OWASP end, but since it's taking a long time for resolution from their end, hence this is just a workaround so that we don't get an issue until issue is fixed from OWASP

Reply:
can you point me another place where we are doing string manipulation on html variable
Whole point of using OWASP was to make sure we don't do string/regex manipulations on html content in performance efficient way, so we are breaking that rule

Reply:
OWASP library is supposed to do all this stuff, but for this particular scenario there seems to be a bug on their end which hasn't been resolved in some time and hence when Rupesh was working on 2547 he came up with this temporary solution
 This entire method is a temporary fix for unbalanced tags issue, which supposedly had to be handle by oWASP library
 @rupeshtelus can add further details of this method and ticket related details to OWASP if required